### PR TITLE
fixes #19162; enable `strictEffects` for v2

### LINKED ISSUE #19162: Experimental features in normal manual instead of experimental manual/undocumented
The experimental features `compiletimeFFI`, ~~`dynamicBindSym`~~ and `vmopsDanger` are not documented in the experimental manual as of 1.6.0 ([full list](https://github.com/nim-lang/Nim/blob/a8611c665f26adbf7f6dd2438359b4c5d930bf86/compiler/options.nim#L189)). Since `vmopsDanger` seems to do very little and `compiletimeFFI` seems to be unfinished, I'm not sure if they need to be documented. `dynamicBindSym` is documented in [macros.bindSym](https://nim-lang.org/docs/macros.html#bindSym%2C%2CBindSymRule) and in [0.19 changelog](https://github.com/nim-lang/Nim/blob/dbf8d0b8946419311ac710b1adc995ad206b52c1/changelogs/changelog_0_19_0.md#compiler-changes) however, which may be enough.

However the following experimental features are documented in the regular manual instead of the experimental manual. Don't know if this is intended.

- `strictEffects` ([manual](https://nim-lang.org/1.6.0/manual.html#effect-system-effectsof-annotation))
- ~~`unicodeOperators` ([manual](https://nim-lang.org/1.6.0/manual.html#lexical-analysis-unicode-operators))~~ 
- ~~`overloadableEnums` ([manual](https://nim-lang.org/1.6.0/manual.html#types-overloadable-enum-field-names))~~

**Update:** Since #19173, only `compiletimeFFI` and `vmopsDanger` are not documented, and `strictEffects` is still in the regular manual, because it may eventually leave experimental status.

### PR DESCRIPTION
**Pull request by ringabout**
fixes https://github.com/nim-lang/Nim/issues/19162

issues tracking

- [x] nimgen, nimterop, prologue  require `nim-regex` to release a new tag https://github.com/nitely/nim-regex/issues/116
- [x] nimquery nimquery\nimquery.nim(749, 42) Error: $demand can raise an unlisted exception: Exception
https://github.com/GULPF/nimquery/pull/19
- [x] telebot asyncmacro.nim(212, 31) Error: 'callbackIter' is not GC-safe as it calls 'unmarshal'
https://github.com/ba0f3/telebot.nim/pull/79
- [x] weave 
https://github.com/mratsim/weave/pull/184
- [x] numericalnim /home/runner/.nimble/pkgs/cdt-#head/cdt/minmax.nim(123, 11) Hint: 'minValueBy' calls routine via hidden pointer indirection
https://github.com/HugoGranstrom/cdt/pull/1
https://github.com/SciNim/numericalnim/pull/31
- [x] parsetoml  /home/runner/work/Nim/Nim/pkgstemp/parsetoml/src/parsetoml.nim(1742, 6) Hint: '==' called by '=='
https://github.com/NimParsers/parsetoml/pull/58
- [x] fusion/tests/tmatching fusion/src/fusion/matching.nim(558, 6) Error: '$' can have side effects
https://github.com/nim-lang/Nim/pull/19380/commits/5842184712c8ce9ab4f2b279e83f6f0a33aad116
- [x] ggplotnim   'addQuoted' calls `.sideEffect` '$'
- [x] yaml  Error: repr(e.parent) can raise an unlisted exception: Exception
https://github.com/flyx/NimYAML/pull/125
- [x] glob
https://github.com/haltcase/glob/pull/58
- [x] docopt Error: 'docopt' is not GC-safe as it calls 'docopt_exc'
https://github.com/docopt/docopt.nim/pull/56

See also https://github.com/nim-lang/Nim/issues/19303

### CODE REVIEW

**Comment by Araq** [comment]
Tests need to be updated but important packages seem to work, so please proceed.

**Comment by ringabout** [comment]
Here is the progress of fixing important packages:

https://github.com/nitely/nim-regex/pull/110

**Comment by ringabout** [comment]
made a PR for arraymancer: https://github.com/mratsim/Arraymancer/pull/549

**Comment by ringabout** [comment]
issues tracking

- [x] nimgen, nimterop, prologue  require `nim-regex` to release a new tag https://github.com/nitely/nim-regex/issues/116
- [x] nimquery nimquery\nimquery.nim(749, 42) Error: $demand can raise an unlisted exception: Exception
https://github.com/GULPF/nimquery/pull/19
- [x] telebot asyncmacro.nim(212, 31) Error: 'callbackIter' is not GC-safe as it calls 'unmarshal'
https://github.com/ba0f3/telebot.nim/pull/79
- [x] weave 
https://github.com/mratsim/weave/pull/184
- [x] numericalnim /home/runner/.nimble/pkgs/cdt-#head/cdt/minmax.nim(123, 11) Hint: 'minValueBy' calls routine via hidden pointer indirection
https://github.com/HugoGranstrom/cdt/pull/1
https://github.com/SciNim/numericalnim/pull/31
- [x] parsetoml  /home/runner/work/Nim/Nim/pkgstemp/parsetoml/src/parsetoml.nim(1742, 6) Hint: '==' called by '=='
https://github.com/NimParsers/parsetoml/pull/58
- [x] fusion/tests/tmatching fusion/src/fusion/matching.nim(558, 6) Error: '$' can have side effects
https://github.com/nim-lang/Nim/pull/19380/commits/5842184712c8ce9ab4f2b279e83f6f0a33aad116
- [ ] ggplotnim   'addQuoted' calls `.sideEffect` '$'
- [x] yaml  Error: repr(e.parent) can raise an unlisted exception: Exception
https://github.com/flyx/NimYAML/pull/125
- [x] glob
https://github.com/haltcase/glob/pull/58
- [x] docopt Error: 'docopt' is not GC-safe as it calls 'docopt_exc'
https://github.com/docopt/docopt.nim/pull/56


See also https://github.com/nim-lang/Nim/issues/19303

File: testament/important_packages.nim [hunk 1]
```diff
@@ -158,11 +158,11 @@ pkg "tiny_sqlite"
...
 pkg "unicodeplus", "nim c -d:release -r tests/tests.nim"
 pkg "unpack"
-pkg "weave", "nimble install -y cligen synthesis;nimble test_gc_arc"
+pkg "weave", "nimble install -y cligen synthesis;nimble test_gc_arc", url = "https://github.com/nim-lang/weave"
 pkg "websocket", "nim c websocket.nim"
 pkg "winim", "nim c winim.nim"
 pkg "with"
 pkg "ws", allowFailure = true
-pkg "yaml", "nim c -r test/tserialization.nim"
+pkg "yaml", "nim c -r test/tserialization.nim", url = "https://github.com/flyx/NimYAML/pull/125", useHead = true
```
Comment (ringabout) [inline testament/important_packages.nim]:
```suggestion
pkg "yaml", "nim c -r test/tserialization.nim", url = "https://github.com/nim-lang/NimYAML", useHead = true
```

File: testament/important_packages.nim [hunk 2]
```diff
@@ -158,11 +158,11 @@ pkg "tiny_sqlite"
 pkg "unicodedb", "nim c -d:release -r tests/tests.nim"
 pkg "unicodeplus", "nim c -d:release -r tests/tests.nim"
 pkg "unpack"
-pkg "weave", "nimble install -y cligen synthesis;nimble test_gc_arc"
+pkg "weave", "nimble install -y cligen synthesis;nimble test_gc_arc", url = "https://github.com/nim-lang/weave"
```
Comment (ringabout) [inline testament/important_packages.nim]:
```suggestion
pkg "weave", "nimble install -y cligen synthesis;nimble test_gc_arc", url = "https://github.com/nim-lang/weave", useHead = true
```

File: testament/important_packages.nim [hunk 3]
```diff
@@ -146,7 +146,7 @@ pkg "strslice"
 pkg "strunicode", "nim c -r --mm:refc src/strunicode.nim"
 pkg "supersnappy"
 pkg "synthesis"
-pkg "telebot", "nim c -o:tbot -r src/telebot.nim"
+pkg "telebot", "nim c -o:tbot -r src/telebot.nim", url = "https://github.com/nim-lang/telebot.nim"
```
Comment (ringabout) [inline testament/important_packages.nim]:
```suggestion
pkg "telebot", "nim c -o:tbot -r src/telebot.nim", url = "https://github.com/nim-lang/telebot.nim, useHead = true"
```

File: testament/important_packages.nim [hunk 4]
```diff
@@ -109,7 +109,7 @@ pkg "nimongo", "nimble test_ci", allowFailure = true
 pkg "nimph", "nimble test", "https://github.com/disruptek/nimph", allowFailure = true
 pkg "nimPNG", useHead = true
 pkg "nimpy", "nim c -r tests/nimfrompy.nim"
-pkg "nimquery"
+pkg "nimquery", url = "https://github.com/nim-lang/nimquery"
```
Comment (ringabout) [inline testament/important_packages.nim]:
```suggestion
pkg "nimquery", url = "https://github.com/nim-lang/nimquery", useHead = true
```

File: testament/important_packages.nim [hunk 5]
```diff
@@ -146,7 +146,7 @@ pkg "strslice"
 pkg "strunicode", "nim c -r --mm:refc src/strunicode.nim"
 pkg "supersnappy"
 pkg "synthesis"
-pkg "telebot", "nim c -o:tbot -r src/telebot.nim"
+pkg "telebot", "nim c -o:tbot -r src/telebot.nim", url = "https://github.com/nim-lang/telebot.nim, useHead = true"
```
Comment (ringabout) [inline testament/important_packages.nim]:
```suggestion
pkg "telebot", "nim c -o:tbot -r src/telebot.nim", url = "https://github.com/nim-lang/telebot.nim", useHead = true
```

File: compiler/sempass2.nim [hunk 6]
```diff
@@ -495,7 +495,7 @@ proc isIndirectCall(tracked: PEffects; n: PNode): bool =
   if n.kind != nkSym:
     result = true
   elif n.sym.kind == skParam:
-    if strictEffects in tracked.c.features:
+    if not isDefined(tracked.c.config, "nimLegacyEffects"):
```
Comment (Araq) [inline compiler/sempass2.nim]: No! Instead add something to options.LegacyFeature

**Pull request merged**